### DISCUSSION TITLE
Courses tab - Ability to show courses that are open for enrollment

### DISCUSSION THREAD

**Discussion by mmanimegalai** [Sunbird-Ed/Community#40] [Enhancement request]
This is a DIKSHA request

Context:
In the “Courses” tab (both on Portal and Mobile), only the Courses that have active batch are shown. However this also includes courses with active batch where enrollment date is over. This does not help users in any way as they can not enroll into it

Required enhancement:
1. In the Courses tab, do not show Courses that have batch with enrollment date crossed
2. Note: The existing logic to show various courses under “My courses” section should be retained as it is

**Comment by pramodkvarma** [comment]
I am surprised that these need code changes at all!! Given APIs are so flexible, why are we not having these UI widgets/components with configurations that allow easy adjustment to these search criteria?

Please do not keep making these tweaks without fixing underlying configurability aspects of UX widgets.

@sharathkashyap @SanthoshVasabhaktula @vrayulu

**Reply by sharathkashyap** [reply]
@pramodkvarma We have investigated the issue from ED side on this issue. Here are our findings:

a) We have capability to send batch.status and batch.enrollmentstatus as filter to ensure that only open and live batches are listed. ED can be configured to send enrollmentEndDate as part of same filter in search query using configuration.

b) Limitation tend to exist in supporting this date attribute as filter parameter in search API call. Currently the attribute is persisted as string. which limits the capability of API to filter results based on date.

We have also created a ticket to Sunbird Knowlg to support this capability quite some time back.
https://project-sunbird.atlassian.net/browse/SB-24090

cc: @luckynvdu

**Reply by pramodkvarma** [reply]
Ok. While we enhance the API, please make sure all such cards/tabs/lists components/widgets have these as configurations. 

E.g. same component should be used and configured with "show all content of type courses", "show only content of type courses which are live", "show upcoming courses", "show content of type events and date next 3 months", etc. All these need only one UX widget that can configure itself given we have flexible generalized APIs.

**Reply by sharathkashyap** [reply]
@pramodkvarma Agreed that UX has to be generalised. Infact, we have a concept of sections which allow us to configure a specific section of the page by keeping search API filters as configuration. We will continue to improve further.

**Comment by sharathkashyap** [comment]
@mmanimegalai https://project-sunbird.atlassian.net/browse/SB-24090 I will be locking this thread. Since specific JIRA issue has been created and acknowledged for the issue.

@rhwarrier Could you help us to prioritise on the ticket.
